M2/Macaulay2/packages/EngineTests/Res.f4.m2

```diff
@@ -738,26 +738,29 @@ TEST ///
   S = ZZ/101[x_1..x_n]
   soc = random(S^1, S^{ -d});
   I = ideal fromDual soc;
+  time C = res I
   time B = betti res I
-  C = res I;
-  time F = res(I, FastNonminimal=>true)
+  time F = res(ideal I_*, FastNonminimal=>true)
   time B == betti(F, Minimize =>true)
   assert(betti F != betti C)
   assert(F =!= C)
 
-  I = ideal I_*;
-  F = res(I, FastNonminimal=>true)
+  F = res(ideal I_*, FastNonminimal=>true)
   assert(B == betti(F, Minimize =>true))
   C = res I;
   assert(B == betti C)
-  
+
+  I = ideal I_*;
   C1 = res(I, Strategy=>1);
   C2 = res(I, Strategy=>2);
   C0 = res(I, Strategy=>0);
   C4 = res(I, Strategy=>4);
   assert(C2 === C1)
   assert(C0 === C1)
-  assert(C4 =!= C1)
+  assert(C4 === C1)
```

Comment:
C1 is the minimal free resolution computed with strategy 1, which is cached at the beginning, therefore future calls to `res` should return it without further computation.

If you look about 9 lines further down, there a non-minimal free resolution C4 is computed first, and at then the minimal free resolution C1 is computed, so now C1 and C4 are different.

In other words, the previous inequality worked because caching was wrong.